Modules/Microsoft365DSC/DSCResources/MSFT_IntuneVPNConfigurationPolicyIOS/MSFT_IntuneVPNConfigurationPolicyIOS.psm1

```diff
@@ -0,0 +1,1333 @@
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $ApplicationSecret,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens,
+        
+        #latest updates
+        [Parameter()]
+        [System.UInt32]
+        $version,
+
+        [Parameter()]
+        [System.String]
+        $loginGroupOrDomain,
+
+        [Parameter()]
+        [System.String]
+        $role,
+
+        [Parameter()]
+        [System.String]
+        $realm,
+
+        [Parameter()]
+        [System.String]
+        $identifier,
+
+        [Parameter()]
+        [System.Boolean]
+        $enablePerApp,
+
+        [Parameter()]
+        [ValidateSet('notConfigured', 'appProxy', 'packetTunnel')]
+        [System.String]
+        $providerType,
+
+        [Parameter()]
+        [System.Boolean]
+        $disableOnDemandUserOverride,
+
+        [Parameter()]
+        [System.Boolean]
+        $disconnectOnIdle,
+
+        [Parameter()]
+        [System.UInt32]
+        $disconnectOnIdleTimerInSeconds,
+
+        [Parameter()]
+        [System.String]
+        $microsoftTunnelSiteId,
+
+        [Parameter()]
+        [System.String]
+        $cloudName,
+
+        [Parameter()]
+        [System.Boolean]
+        $strictEnforcement,
+
+        [Parameter()]
+        [System.String]
+        $userDomain
+    )
+
+    try
+    {
+        $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
+            -InboundParameters $PSBoundParameters
+    }
+    catch
+    {
+        Write-Verbose -Message $_
+    }
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $currentInstance = Get-TargetResource @PSBoundParameters
+
+    $BoundParameters = Remove-M365DSCAuthenticationParameter -BoundParameters $PSBoundParameters
+
+    #proxy and server values need converting before new- / update- cmdlets will accept parameters
+    #creating hashtables now for use later in both present/present and present/absent blocks
+    $allTargetValues = Convert-M365DscHashtableToString -Hashtable $BoundParameters
+    if ($allTargetValues -match '\bserver=\(\{([^\)]+)\}\)') {$serverBlock = $matches[1]}
```

Comment:
As previously mentioned, conditions and loops should have the brackets in their own line and the inside block properly indented. I see the same mistake several times in the chunks right below this.

Reply:
Sorry, bad habit of mine. Fixed.

Reply:
@dannyKBjj In my own projects, I always put the opening bracket on the same line. But for the sake of consistency here, we all put them on a new line. Doesn't need to be a "bad" habit, just consistency 👍

Reply:
Thanks for chiming in, I also do that in all my projects and use tabs (8 spaces) instead of spaces, but there's a different coding style here that should be consistent.